File: JS/1.1-Number Systems/PrintAllDivisorsOfN.cpp
function printAllDivisorsOfN(N) {
  const divisors = []

  for (let i = 2; i <= Math.sqrt(N); i++) {
    if (N % i == 0) {
      divisors.push(i)
      if (i !== N / i) {
        divisors.push(N / i)
      }
    }
  }

  divisors.sort((a, b) => a - b) // Sort the divisors in ascending order
  divisors.forEach((divisor) => console.log(divisor))
}

printAllDivisorsOfN(100)

File: JS/1.1-Number Systems/PrintAllDivisorsOfN.js
/** Finding All the Divisors of number N
 * TC : O(sqrt(N))
 * SC : O(N)
 */
function allDivisorsOfN(N) {
  let result = []
  for (let i = 1; i <= Math.sqrt(N); i++) {
    if (N % i == 0) {
      result.push(i)
      if (i !== N / i) result.push(N / i)
    }
  }
  return result
}

console.log(allDivisorsOfN(8)) // Output: [1, 8, 2, 4]
console.log(allDivisorsOfN(28)) // Output: [1, 28, 2, 14, 4, 7]
console.log(allDivisorsOfN(1)) // Output: [1]
